packages/jellyfish-api-core/__tests__/category/wallet/listTransactions.test.ts

```diff
@@ -0,0 +1,230 @@
+import { BigNumber } from '@defichain/jellyfish-json'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+describe('listTransactions', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  const address = 'mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU'
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should listTransactions', async () => {
+    await Promise.all(Array.from({ length: 10 }).map(async (_, i) => {
+      return await client.wallet.sendToAddress(address, 0.0001)
+    }))
+    await container.generate(1, address)
+
+    const inWalletTransactions = await client.wallet.listTransactions({})
+
+    expect(inWalletTransactions.length).toStrictEqual(10)
+
+    for (const inWalletTransaction of inWalletTransactions) {
+      expect(inWalletTransaction).toMatchObject({
```

Comment:
Is there a reason why we prefer `toMatchObject` here over `toStrictEqual`?

Reply:
Here are the specifications on what attributes are returned by the `listtransactions` function on DeFiChain

https://github.com/DeFiCh/ain/blob/master/src/wallet/rpcwallet.cpp
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/15076300/220551976-6430eee8-3cd2-4b2c-a800-5e43828a2ed0.png">

When running the tests, sometimes there are additional optional returned parameters that are not included in the specifications above:
- `generated` -> boolean 
- `walletconflicts` -> Array

These parameters appear in some responses, and not others.

Using `toStrictEqual` may fail tests here, the extra parameters `generated` & `walletconflicts` are not always included in the response. 

However, `toMatchObject` is defined as "check that a JavaScript object matches a subset of the properties of an object"

In this case, we can just compare the response result if it matches a subset of the specification parameters from the DeFiChain repository

Reply:
If anyone from the DeFi chain team knows when & why the 2 extra parameters appear, feel free to add on here so I can modify my tests to use `toStrictEqual`

Reply:
Ah okay that makes sense 

I guess the RPC doc needs updating in `ain` then

Reply:
do you know who should i ask here? @eli-lim 

I'll probably raise a ticket to the `ain` repo?

Reply:
Sure, just raise a ticket

In the mean time, let's wait for others to review this PR